File: tests/src/main/java/com/smartamenities/backend/dto/AmenityResponse.java
package com.smartamenities.backend.dto;

import com.smartamenities.backend.model.AmenityType;
import com.smartamenities.backend.model.AmenityStatus;
import com.smartamenities.backend.model.AmenityStatusReason;
import com.smartamenities.backend.model.OccupancyStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for responding to GET /amenities endpoint.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AmenityResponse {
    private String id;
    private String name;
    private AmenityType amenityType;
    private String level;
    private double latitude;
    private double longitude;
    private Double distanceMeters;
    private AmenityStatus status;
    private AmenityStatusReason statusReason;
    private Integer waitTimeMinutes;
    private Integer stallsAvailable;
    private OccupancyStatus occupancyStatus;
    private Boolean selectable;
}

File: Mobile/app/src/main/java/com/smartamenities/app/model/AmenityResponse.kt
package com.smartamenities.app.model

/**
 * Amenity data returned by GET /amenities endpoint.
 * 
 * distanceMeters is route-network distance (not haversine) computed by backend
 * using the same graph routing engine as POST /route.
 * Android treats this as authoritative and does NOT recompute distance.
 */
data class AmenityResponse(
    val id: String,
    val name: String,
    val amenityType: String,
    val level: String,
    val latitude: Double,
    val longitude: Double,
    val distanceMeters: Double? = null,  // Backend-computed route distance; do not override in Android
    val status: String? = null,
    val statusReason: String? = null,
    val waitTimeMinutes: Int? = null,
    val stallsAvailable: Int? = null,
    val occupancyStatus: String? = null,
    val selectable: Boolean? = null
)

